### DISCUSSION TITLE
Error - ldapmsg invalid

### DISCUSSION THREAD

**Discussion by jOeOOeOj** [lldap/lldap#920] [Q&A] [answered]
Hi,

i am trying to get lldap done. But everytime i connect i get this message.

> [error]: [LDAP] Service Error: while handling incoming messages: while receiving LDAP op: ldapmsg invalid

I have one admin and on user. The rest is just out of the box with the docker compose file.

Where can i search?

Thx

```
0x0 = ldap_unbind(ld);
ld = ldap_open("192.168.2.45", 389);
Established connection to 192.168.2.45.
Retrieving base DSA information...
Getting 1 entries:
Dn: (RootDSE)
defaultNamingContext: dc=test,dc=loc; 
isGlobalCatalogReady: false; 
namingContexts: dc=test,dc=loc; 
objectClass: top; 
supportedControl: 
supportedExtension: 1.3.6.1.4.1.4203.1.11.1; 
supportedFeatures: 1.3.6.1.4.1.4203.1.5.1; 
supportedLDAPVersion: 3; 
vendorName: LLDAP; 
vendorVersion: lldap_0.5.0; 

0 = ldap_set_option(ld, LDAP_OPT_ENCRYPT, 1)
res = ldap_bind_s(ld, NULL, &NtAuthIdentity, NEGOTIATE (1158)); // v.3
	{NtAuthIdentity: User='TestUser'; Pwd=<unavailable>; domain = 'test.loc'}
Error <52>: ldap_bind_s() failed: Nicht verfügbar.
Server error: <empty>

0 = ldap_set_option(ld, LDAP_OPT_ENCRYPT, 1)
res = ldap_bind_s(ld, NULL, &NtAuthIdentity, NEGOTIATE (1158)); // v.3
	{NtAuthIdentity: User='TestUser'; Pwd=<unavailable>; domain = 'dc=test,dc=loc'}
Error <52>: ldap_bind_s() failed: Nicht verfügbar.
Server error: <empty>
```

**Comment by nitnelave** [comment]
How are you trying to connect to the server? What logs are these from? 

LLDAP doesn't really support LDAP viewers, they require a much wider support of the protocol than we implement.

**Reply by jOeOOeOj** [reply]
The logs are from ldp.exe from the Windows Server Manager Tools.
I also tried Nettools witch is much bigger. I know that this tool could be "to big" so i tried ldp.exe. Maybe this is also "to big" but i dont found an information witch tool i can use to check functionaltiy from a windows pc. Can you tell me how i can check the function from my windows pc?
Thx

**Reply by jOeOOeOj** [reply, marked as answer]
I found this tool witch can connect to lldap from windows.
[https://sourceforge.net/p/ldapadmin/discussion/305548](url)

Maybe it helps someone else.
